plugins/devtools-backend/src/service/router.ts

```diff
@@ -33,24 +33,38 @@ import { errorHandler } from '@backstage/backend-common';
 import express from 'express';
 import { getBearerTokenFromAuthorizationHeader } from '@backstage/plugin-auth-node';
 import { createPermissionIntegrationRouter } from '@backstage/plugin-permission-node';
+import { SignalService } from '@backstage/plugin-signals-node';
 
 /** @public */
 export interface RouterOptions {
   devToolsBackendApi?: DevToolsBackendApi;
   logger: Logger;
   config: Config;
   permissions: PermissionEvaluator;
+  signalService?: SignalService;
 }
 
 /** @public */
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, config, permissions } = options;
+  const { logger, config, permissions, signalService } = options;
 
   const devToolsBackendApi =
     options.devToolsBackendApi || new DevToolsBackendApi(logger, config);
 
+  if (signalService) {
+    // Publish info periodically using the signal service
+    setInterval(async () => {
```

Comment:
Not sure if this is something you in the end want to ship as part of this PR? If it is we'd need to clean it up a little bit. We'd need to make sure we clean up this interval on shutdown so the backend doesn't keep trying to send signals during shutdown.